src/rules/require-properties.ts

```diff
@@ -1,30 +1,32 @@
-import type { PackageJsonRuleModule } from "../createRule.js";
+import {
+	CreateRequirePropertyRuleOptions,
+	createSimpleRequirePropertyRule,
+} from "../utils/createSimpleRequirePropertyRule.js";
 
-import { createSimpleRequirePropertyRule } from "../utils/createSimpleRequirePropertyRule.js";
-
-// List of all properties we want to create require- rules for,
-// in the format [propertyName, isRecommended]
-const properties = [
-	["author", false],
-	["description", true],
-	["engines", false],
-	["files", false],
-	["keywords", false],
-	["name", true],
-	["type", true],
-	["types", false],
-	["version", true],
-	// TODO: More to come!
-] satisfies [string, boolean][];
-
-/** All require- flavor rules */
-export const rules = properties.reduce<Record<string, PackageJsonRuleModule>>(
-	(acc, [propertyName, isRecommended]) => {
-		acc[`require-${propertyName}`] = createSimpleRequirePropertyRule(
-			propertyName,
-			isRecommended,
-		);
-		return acc;
+const properties: {
+	name: string;
+	options?: CreateRequirePropertyRuleOptions;
```

Comment:
Could we model this after `valid-properties`?  (That didn't exist when you first authored this PR, but has since been merged).  It has a very similar purpose, but managed this properties array in a slightly more ergonomic way, where name is still the first element of each array.